lib/path/default.nix

```diff
@@ -137,6 +141,84 @@ in /* No rec! Add dependencies on this file at the top. */ {
     subpathInvalidReason value == null;
 
 
+  /* Join subpath strings together using `/`, returning a normalised subpath string.
+
+    Like `concatStringsSep "/"` but safer, specifically:
+
+    - All elements must be valid subpath strings, see `lib.path.subpath.isValid`
+
+    - The result gets normalised, see `lib.path.subpath.normalise`
```

Comment:
Not really possible since it's not rendered as markdown and we don't know the link of this manual in the end. I'd like to defer this to when we have proper markdown rendering